src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -126,11 +128,11 @@ protected override void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonW
 
             if (value == null)
             {
-                writer.WriteNull(key);
+                writer.WriteNull(key!);
```

Comment:
We already have a `Debug.Assert(key != null) `above. If you mark `ThrowInvalidOperationException_SerializerDictionaryKeyNull` as `DoesNotReturn`, you wouldn't need to add these `!`.

Same with line 135.

Reply:
right, removed